crates/ironrdp-connector/src/lib.rs

```diff
@@ -97,6 +100,38 @@ pub struct Config {
     pub no_server_pointer: bool,
     /// If true, the INFO_AUTOLOGON flag is set in the [`ironrdp_pdu::rdp::ClientInfoPdu`].
     pub autologon: bool,
+    // for kerberos authentication
+    pub sspi_config: Option<SspiConfig>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SspiConfig {
+    pub kdc_url: Option<url::Url>,
+    pub hostname: Option<String>,
+}
+
+impl SspiConfig {
+    pub fn new(kdc_proxy_url: Option<impl Into<String>>, hostname: Option<impl Into<String>>) -> ConnectorResult<Self> {
+        let kdc_url: Option<url::Url> = match kdc_proxy_url {
+            Some(inner) => {
+                let url_str: String = inner.into();
+                let url = url::Url::parse(&url_str).map_err(|_| general_err!("kdc url cannot be parsed"))?;
+                Some(url)
+            }
+            None => None,
+        };
+        let hostname: Option<String> = hostname.map(|inner| inner.into());
+        Ok(Self { kdc_url, hostname })
```

Comment:
Can be greatly simplified (along above comment):
```suggestion
        let kdc_url = kdc_proxy_url.and_then(|url| url::Url::parse(&url).ok());
        Ok(Self { kdc_url, hostname })
```